File: Data Structures & Algorithms/binary-tree-diameter/submission-0.swift
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     var val: Int
 *     var left: TreeNode?
 *     var right: TreeNode?
 *     init(_ val: Int) {
 *         self.val = val
 *         self.left = nil
 *         self.right = nil
 *     }
 * }
 */

class Solution {
    func diameterOfBinaryTree(_ root: TreeNode?) -> Int {
        var count = 0

        func dfs(_ root: TreeNode?) -> Int{
            guard let root = root else{return 0}
            let l = dfs(root.left)
            let r = dfs(root.right)
            count = max(count, l + r)
            return 1 + max(l, r)
        }
        dfs(root)
        return count
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        def rec(root, h):
            if not root:
                return None
            if len(res) < h+1:
                res.append([])
            res[h].append(root.val)
            rec(root.left, h+1)
            rec(root.right, h+1)
        rec(root, 0)
        
        for i in range(len(res)):
            if res[i]:
                res[i] = res[i][-1]
            else:
                del res[i]
            
        return res